### ISSUE TITLE
AI substracts points from Pollen Puff incorrectly when targeting enemies

### ISSUE THREAD

**Issue by AsparagusEduardo** [rh-hideout/pokeemerald-expansion#2611]
Found Alex N. on RHH's discord.

Its current structure causes Pollen Puff to always substract points when targeting enemies.
```c
        case EFFECT_HIT_ENEMY_HEAL_ALLY:    // pollen puff
            if (IsTargetingPartner(battlerAtk, battlerDef))
            {
                if (gStatuses3[battlerDef] & STATUS3_HEAL_BLOCK)
                    return 0;
                if (AtMaxHp(battlerDef))
                    score -= 10;
                else if (gBattleMons[battlerDef].hp > gBattleMons[battlerDef].maxHP / 2)
                    score -= 5;
                break;
            }
            // fallthrough
        case EFFECT_HEAL_PULSE: // and floral healing
            if (!IsTargetingPartner(battlerAtk, battlerDef)) // Don't heal enemies
            {
                score -= 10;
            }
            else
            {
                if (AtMaxHp(battlerDef))
                    score -= 10;
                else if (gBattleMons[battlerDef].hp > gBattleMons[battlerDef].maxHP / 2)
                    score -= 5;
            }
            break;
```
Here's my proposal of a fix to not only correct this but also fix the AI trying to heal allies under Heal Block:
```c
        case EFFECT_HEAL_PULSE: // and floral healing
            if (!IsTargetingPartner(battlerAtk, battlerDef)) // Don't heal enemies
            {
                score -= 10;
                break;
            }
            // fallthrough
        case EFFECT_HIT_ENEMY_HEAL_ALLY:    // pollen puff
            if (IsTargetingPartner(battlerAtk, battlerDef))
            {
                if (gStatuses3[battlerDef] & STATUS3_HEAL_BLOCK)
                    return 0;
                if (AtMaxHp(battlerDef))
                    score -= 10;
                else if (gBattleMons[battlerDef].hp > gBattleMons[battlerDef].maxHP / 2)
                    score -= 5;
            }
            break;
```

**Comment by Syren-A** [comment]
As discussed on discord pollen puff got a -10 score because it didn't want to heal the player. 
![pollen_puff_01](https://user-images.githubusercontent.com/93446519/215295900-b31e723a-d6b3-42b4-bc3a-b152c60fbafe.png)
With the changes that Edu made the scores are fixed.
![pollen_puff_02](https://user-images.githubusercontent.com/93446519/215296053-de054a81-2fbc-4a39-883a-8350ca721f04.png)

There is a different issue with the partner healing moves but I think EFFECT_HEAL_PULSE is just missing/was never planned to be added in AI_DoubleBattle.

**Comment by DizzyEggg** [comment]
This got fixed with #2648
